llvm/test/tools/llvm-dlltool/arm64ec.test

```diff
@@ -53,3 +66,8 @@ EXPORTS
 LIBRARY test.dll
 EXPORTS
     otherfunc
+
+#--- test3.def
+LIBRARY test.dll
+EXPORTS
+    func == efunc
```

Comment:
Hmm, isn't this the case that end up using `EXPORTAS` already, since https://github.com/llvm/llvm-project/pull/98229? Would it be clearer to test a case here, where you'd have a separate `efunc` entry, so that it shows that even if we have the target exported, we'd still use the `EXPORTAS` type instead of weak aliases?

Reply:
Good point, that's a more interesting case. I pushed more tests.

FWIW, we were not using weak alias path in that test, but for a wrong reason: we were comparing mangled name to demangled one (and generally skipping mangling fixup). That's no longer a problem with this PR as there are no deferred on EC target. Looking at it again we should probably take `ExportName` into account when filling `RegularImports` map anyway for corner cases with a mixture of mingw-style aliases and explicit EXPORTAS .